doc/kernel/threads/lifecycle.rst

```diff
@@ -191,6 +205,43 @@ The following code has the same effect as the code segment above.
                     my_entry_point, NULL, NULL, NULL,
                     MY_PRIORITY, 0, K_NO_WAIT);
 
+User Mode Constraints
+---------------------
+
+This section only applies if :option:`CONFIG_USERSPACE` is enabled, and a user
+thread tries to create a new thread. The :c:func:`k_thread_create()` API is
+still used, but there are additional constraints:
+
+* The child thread object must be in an uninitailized state, i.e. it is
+  not currently running.
+
+* The child's stack object must also be uninitialized.
+
+* The stack size parameter passed in must be equal to or less than the
+  bounds of the stack object when it was declared.
+
+* The :c:macro:`K_USER` option must be used, as user threads can only create
+  other user threads.
+
+* The :c:macro:`K_ESSENTIAL` option must not be used, user threads may not be
+  considered essential threads.
+
+* The priority of the child thread must be a valid priority value, and equal to
+  or worse than the parent thread.
+
+* The total number of threads created must not exceed the maximum number of
+  threads that can exist in the system; this value is controlled by the
+  :option:`CONFIG_MAX_THREAD_BYTES` option.
+
+Dropping Permissions
+====================
+
+If :option:`CONFIG_USERSPACE` is enabled, a thread running in supervisor mode
+may perform a one-way transition to user mode using the
+:cpp:func:`k_thread_user_mode_enter()` API. This is a one-way operation which
+will reset and zero the thread's stack memory. The thread will be marked
+as non-essential.
```

Comment:
if thread stack is cleared then will it restart from therad_entry()?

Reply:
yes, correct

Reply:
if you look at the API for k_thread_user_mode_enter() it takes an entry point and 3 parameters